src/proxies/rolesapi.js

```diff
@@ -0,0 +1,82 @@
+const platformClient = require('purecloud-platform-client-v2');
+const retry = require('@lifeomic/attempt').retry;
+
+let rolesMap = new Map();
+
+/*
+    The getAuthorizationRoleByLogicalName() will look up a role based on its logical name from GenesysCloud from Genesys Cloud 
+*/
+const getAuthorizationRoleByLogicalName = async (logicalName) => {
+  let opts = {
+    name: logicalName,
+  };
+
+  let apiInstance = new platformClient.AuthorizationApi();
+
+  try {
+    const results = await apiInstance.getAuthorizationRoles(opts);
+
+    const role = {
+      id: results.entities[0].id,
+      name: results.entities[0].name,
+    };
+
+    if (results != null) {
+      rolesMap[role.name] = role;
+      return {...role};
+    }
+
+    return null;
+  } catch (err) {
+    console.log(
+      `Error while retrieving role with name: ${logicalName}: ${JSON.stringify(
+        err,
+        null,
+        '\t'
+      )}`
+    );
+    return null;
+  }
+};
+
+/* Retrieves a role based on is role name*/
+const getRoleByName = async (roleName) => {
+  const results = rolesMap[roleName];
+  if (results != null) {
+    return results;
+  } else {
+    const resultsLookup = await getAuthorizationRoleByLogicalName(roleName);
+    return resultsLookup;
+  }
+};
+
+/* Return a list of roles for all roles returned*/
+const getRoleIds = () => {
+  const roleIds = new Set(Object.keys(rolesMap).map((key) => rolesMap[key].id));
```

Comment:
probably simpler to just do this
```suggestion
  return Object.values(rolesMap).map((value) => value.id));
```

Reply:
yeah.  I actually was doing something different before I dont think i need the dedup any more.  Going to change it.